src/home/homework-luks.c

```diff
@@ -63,6 +63,7 @@
                 _x > UINT64_MAX - 4095U ? UINT64_MAX : (_x + 4095U) & ~UINT64_C(4095); \
         })
 
+static int resize_image_loop(UserRecord *h, HomeSetup *setup, uint64_t old_image_size, uint64_t new_image_size, uint64_t *ret_image_size);
```

Comment:
any reason to avoid moving the caller further down?

Reply:
to keep this together with resize_image_fs(). i.e. the two functions are closely related: one is the bisection loop when growing, i.e. when we first to grow the backing file to the max the underlying fs allows, and the other is the bisection loop when shrinking, i.e. when we first try to shrink the fs to minimum it allows. I think keeping them together, along with the rest of the resize code makes sense.